libcxx/include/__functional/move_only_function_impl.h

```diff
@@ -133,20 +117,20 @@ class _LIBCPP_MOVE_ONLY_FUNCTION_TRIVIAL_ABI move_only_function<_ReturnT(
   _LIBCPP_HIDE_FROM_ABI void __construct(_Args&&... __args) {
     static_assert(is_constructible_v<decay_t<_Func>, _Func>);
 
-    using _FuncWraps = __function_wrappers<_Func>;
-    using _UnRefFunc = remove_reference_t<_Func>;
+    using _FuncWraps  = __function_wrappers<_Func>;
+    using _StoredFunc = decay_t<_Func>;
 
     __call_ = &_FuncWraps::__call;
     if constexpr (!is_trivially_destructible_v<decay_t<_Func>>) {
       __destroy_ = &_FuncWraps::__destroy;
     }
 
     if constexpr (__fits_in_buffer<_Func>) {
-      std::construct_at(reinterpret_cast<_UnRefFunc*>(__buffer_.data()), std::forward<_Args>(__args)...);
+      std::construct_at(reinterpret_cast<_StoredFunc*>(__buffer_.data()), std::forward<_Args>(__args)...);
     } else {
       unique_ptr<std::byte[]> __ptr{
           static_cast<std::byte*>(::operator new(sizeof(_Func), std::align_val_t(alignof(_Func))))};
-      std::construct_at(reinterpret_cast<_UnRefFunc*>(__ptr.get()), std::forward<_Args>(__args)...);
+      std::construct_at(reinterpret_cast<_StoredFunc*>(__ptr.get()), std::forward<_Args>(__args)...);
       std::construct_at(reinterpret_cast<std::byte**>(__buffer_.data()), __ptr.release());
```

Comment:
Throughout, `reinterpret_cast` is correct, but `static_cast` would be simpler. You can `static_cast` from `void*` to any-other-pointer-type, and for passing around these untyped memory addresses you _should_ be using `void*` rather than `std::byte*` (per my strong recommendation above).